File: frontend/src/types/work-calendar.ts
export interface WorkCalendarDay {
  id: number
  work_date: string
  day_type: 'holiday' | 'workday'
  name: string
  source?: string
  created_at: string
  updated_at: string
}

export interface WorkCalendarDayPayload {
  day_type: 'holiday' | 'workday'
  name: string
  source?: string
}

export interface ScheduleDayMeta {
  kind: 'workday' | 'holiday' | 'weekend'
  name?: string
}

File: backend/app/schemas/work_calendar.py
from datetime import date, datetime
from typing import Literal

from pydantic import Field

from app.schemas.common import ORMModel


class WorkCalendarDayUpsert(ORMModel):
    day_type: Literal["holiday", "workday"]
    name: str = Field(min_length=1, max_length=100)
    source: str | None = Field(default=None, max_length=255)


class WorkCalendarDayResponse(WorkCalendarDayUpsert):
    id: int
    work_date: date
    created_at: datetime
    updated_at: datetime
